checks.d/storm_rest_api.py

```diff
@@ -0,0 +1,301 @@
+        topologies_re = re.compile(instance.get('topologies', '(.*)'))
+
+        task_tags_from_json = {}
+        tag_file = instance.get('task_tags_file', None)
+        if tag_file is not None:
+            with open(tag_file, 'r') as f:
+                task_tags_from_json = json.load(f)
+        task_tags = instance.get('task_tags', {}).copy()
+        task_tags.update(task_tags_from_json)
+
+        raw_whitelist = set(instance.get('executor_details_whitelist', []))
+        executor_details_whitelist = [re.compile(regex) for regex in raw_whitelist]
+
+        return StormConfig(
+            metric_prefix=instance.get('metric_prefix', None),
+            url=url,
+            timeout=timeout,
+            topology_timeout=topology_timeout,
+            tags=instance.get('tags', []),
+            executor_details_whitelist=executor_details_whitelist,
+            task_tags=task_tags,
+            cache_file=cache_file,
+            cache_staleness=instance.get('cache_staleness', self.DEFAULT_STALENESS),
+            topologies=topologies_re,
+        )
+
+    def metric(self, config, name):
+        if config.metric_prefix is not None:
+            return "%s.%s" % (config.metric_prefix, name)
+        else:
+            return name
+
+    def check(self, instance):
+        config = self.instance_config(instance)
+
+        check_status = AgentCheck.OK
+        check_msg = 'Everything went well'
+        check_tags = [
+            'url:' + config.url,
+        ] + config.tags
+
+        details = None
+        with open(config.cache_file, 'r') as cache_f:
+            details = json.load(cache_f)
+
+        oldest_acceptable_cache_time = time.time() - config.cache_staleness
+        if details['status'] == 'success' and details['updated'] > oldest_acceptable_cache_time:
+            data = details['data']
+            self.report_cluster(config, data['cluster'])
+            self.report_supervisors(config, data['supervisors'])
+            self.report_topologies(config, data['topologies'])
+            for topology_detail in data['topology_details']:
+                self.report_topology(config, topology_detail['name'], topology_detail['topology'])
+                for executor_details in topology_detail['component_details']:
+                    self.report_executor_details(config, executor_details)
+        else:
+            check_status = AgentCheck.CRITICAL
+            if details['status'] != 'success':
+                check_msg = "Could not connect to URL %s with timeout %d" % (failure.url, failure.timeout)
+            else:
+                check_msg = "Cache is too stale: %d vs. expected minimum %d" % (details['updated'], oldest_acceptable_cache_time)
+
+        self.service_check(
+            self.metric(config, 'storm.rest.cached_data_ok'),
+            check_status,
+            message=check_msg,
+            tags=check_tags)
+
+    def report_cluster(self, config, cluster):
```

Comment:
It's a very common pattern in DD agents to represent this as a dict:
```python
{
  'slotsUsed': 'cluster.slots_used_count',
…
}
```

Then `iteritems()` over that and put them all in. That would make this shorter, and a bit easier on the eye. (And in other places, since we should probably follow DD's guidelines even if this isn't their repo.)